### DISCUSSION TITLE
Use folder as category, but only 1st level?

### DISCUSSION THREAD

**Discussion by maphew** [getpelican/pelican#3095] [Q&A] [answered]
Is it possible to use the setting `USE_FOLDER_AS_CATEGORY = True` but somehow restrict it to only the first level? (or 2nd, or ...)

For example given content is arranged like:

    ./howto/fix_empty_email_img.md
    ./howto/Windows/enable_symlinks_no_admin.md

I only want `howo` as a category, however with the above I get both `howto` and `Windows` as categories.

**Comment by cjhabs** [comment]
If you set "USE_FOLDER_AS_CATEGORY = False" in pelicanconf.py and set "category: howto" in your frontmatter that should do the trick. You do need to have the category in every document but I find that to not really be an inconvenience.

**Reply by maphew** [reply]
Haha! You and I are in different places. It was precisely the inconvenience of having to edit all the frontmatter categories that led to the question! I would probably feel differently if it were just a matter of me adding the category as I write each new piece. However at the moment I am migrating from another generator so there are a lot of documents to deal with. The solution I came up with is a bit ugly, but at least it let me get to another stage:

```python
'''Quick & dirty add 'category' to yaml front matter using top-most
parent folder from root as category name.

Find `tags:.*` and replace with `category: {top-folder-of-path} \n tags:.*`

WARNING: don't run this unless content is under source control!! It will wipe
out file contents if encounters something it doesn't know how to handle.

Sources:
 - https://stackoverflow.com/questions/10507230/insert-line-at-middle-of-file-with-python
 - ChatGPT
 - https://stackoverflow.com/a/69508474/14420
'''
import os
import glob
import fileinput

root = r"C:\path\to\\pcan\content"

match = 'tags:'

os.chdir(root)
print(os.path.abspath(os.path.curdir))
for f in (glob.glob('*/*.md', recursive=True)):
    print(f'file: {f}')
    cat = os.path.basename(os.path.abspath(os.path.join(f, os.pardir)))
    print(f'category: {cat}')
    skip = False
    for line in fileinput.input(f, inplace=True, encoding='utf-8'):
        if line.startswith('category:'):
            skip = True
        if not skip and line.startswith(match):
            line = f'category: {cat}\n{line}'
        print(line, end='')
```

It worked for me because I could rely on all articles having a `tags:` frontmatter. It's not safe as a general solution.

**Comment by jefdriesen** [comment, marked as answer]
You can use this in your configuration:
```
# Use first level folder as category
USE_FOLDER_AS_CATEGORY = False
PATH_METADATA = '(?P<category>[^/]*)/.*'
```
